drivers/fuel_gauge/max17048/max17048.c

```diff
@@ -0,0 +1,229 @@
+/* max17048.c - Driver for max17048 battery fuel gauge */
+
+/*
+ * Copyright (c) 2023 Alvaro Garcia Gomez <maxpowel@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max17048
+
+#include "max17048.h"
+
+#include <zephyr/drivers/fuel_gauge.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+
+#include <zephyr/logging/log.h>
+
+
+LOG_MODULE_REGISTER(MAX17048);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "MAX17048 driver enabled without any devices"
+#endif
+
+struct max17048_data {
+	float charge;
+	float voltage;
+	uint16_t time_to_full;
+	uint16_t time_to_empty;
+	bool charging;
+};
+
+/* Read buffer */
+unsigned char max17048_buffer[2];
```

Comment:
Since we only use this in the `max17048_read_register()` function, let's make it local to that function.